File: client-clojure/src/sirqul_io_t_platform/specs/connected_friend.clj
(ns sirqul-io-t-platform.specs.connected-friend
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset :refer :all]
            )
  (:import (java.io File)))


(def connected-friend-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :display) string?
   (ds/opt :avatar) asset-spec
   (ds/opt :avatarURL) string?
   (ds/opt :platform) string?
   (ds/opt :platformId) string?
   (ds/opt :invited) boolean?
   (ds/opt :locationDescription) string?
   })

(def connected-friend-spec
  (ds/spec
    {:name ::connected-friend
     :spec connected-friend-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ConnectedFriend.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ConnectedFriend._

case class ConnectedFriend (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
display: Option[String],
avatar: Option[Asset],
avatarURL: Option[String],
platform: Option[String],
platformId: Option[String],
invited: Option[Boolean],
locationDescription: Option[String])

object ConnectedFriend {
  import DateTimeCodecs._

  implicit val ConnectedFriendCodecJson: CodecJson[ConnectedFriend] = CodecJson.derive[ConnectedFriend]
  implicit val ConnectedFriendDecoder: EntityDecoder[ConnectedFriend] = jsonOf[ConnectedFriend]
  implicit val ConnectedFriendEncoder: EntityEncoder[ConnectedFriend] = jsonEncoderOf[ConnectedFriend]
}
